### PR TITLE
Tests for reading windowed side input from resumed SDF call

### PR DESCRIPTION
**Pull request by jkff**
R: @tgroh

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java [hunk 1]
```diff
@@ -287,9 +288,94 @@ public void testWindowedSideInput() throws Exception {
...
+                Create.timestamped(
+                    TimestampedValue.of("a", new Instant(0)),
+                    TimestampedValue.of("b", new Instant(2))))
+            .apply("window 2", Window.<String>into(FixedWindows.of(Duration.millis(2))))
+            .apply("singleton", View.<String>asSingleton());
+
+    PCollection<KV<String, Iterable<Integer>>> res =
+        mainInput
+            .apply(
+                ParDo.of(new SDFWithMultipleOutputsPerBlockAndSideInput(sideInput))
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java]: How does this ensure that the SDF is resumed from a checkpoint?

Comment (jkff) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java]: SDFWithMultipleOutputsPerBlock outputs significantly more items than the standard OutputAndTimeBoundedProcessElementInvoker allows per single call.

At some point I'll reintroduce ProcessContinuation and refactor this test and some others to force checkpointing, but for now this seems better than not having this test :)

Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java]: Perhaps you could output whenever you resume from a checkpoint and assert that the result PCollection is non-empty?

Comment (jkff) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java]: Done, great idea, thanks!

**Pull request closed without merging**